### DISCUSSION TITLE
Pipe input to CLI?

### DISCUSSION THREAD

**Discussion by ghost** [mockoon/mockoon#1106] [Help] [answered]
Can I pipe input to the cli so I can have a script alter the JSON environment file prior to starting the server?

**Comment by 255kb** [comment, marked as answer]
Piping to the CLI is not supported as far as I know. 
But what prevents you to run two commands, like a `sed` followed by the CLI `start` command?

**Reply by ghost** [reply]
That was my backup thought, just passing the environment file through something like `sed` into a temporary file that is fed to the start command. Just figured before I had my CI setup creating temporary files I'd see if I could just pipe stdio to the cli. Thanks for the reply, this is good enough.

**Reply by 255kb** [reply]
If the data file is already local indeed there is no overhead, but if you load it from a URL then yes, you may have to create a temporary file.
